internal/source/source.go

```diff
@@ -47,36 +46,23 @@ func loadConfigFile(filename string) (_ *Config, retErr error) {
 	return config, nil
 }
 
-func gatherSourceFilenames(root string, depth int) ([]string, error) {
+func gatherSourceFilenames(root string) ([]string, error) {
 	var filenames []string
 	if err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
 		if err != nil {
 			return err
 		}
 		if d.IsDir() {
-			count := strings.Count(strings.TrimPrefix(path, root), string(os.PathSeparator))
-			if count == 0 {
-				return nil
-			} else if count > depth {
-				return fs.SkipDir
-			}
-			files, err := os.ReadDir(path)
-			if err != nil {
-				return err
-			}
-			var found bool
-			for _, file := range files {
-				if file.Name() == "source.yaml" {
-					found = true
-					break
+			switch d.Name() {
+			case ".": // Allow relative path to current directory
+			case "cmd", "internal", "tests":
+				return filepath.SkipDir
+			default:
+				if strings.HasPrefix(d.Name(), ".") {
+					return filepath.SkipDir
 				}
 			}
-			if !found {
-				return fmt.Errorf("%w: %s", ErrSourceFileNotFound, path)
-			}
-			return nil
-		}
-		if d.Name() == "source.yaml" {
+		} else if d.Name() == "source.yaml" {
```

Comment:
Relaxed checks in here to gather all of the `source.yaml` files at the top-level directory.

Reply:
Nice cleanup, now that I look at the existing code it was a bit of a mess.